docs/source/architecture/journal.rst

```diff
@@ -0,0 +1,379 @@
+   has expired, but could be on any other criteria the consensus algorithm has
+   for determining if it is time to publish a block. When this returns true the
+   BlockPublisher will proceed in creating the block.
+3. finalize_block - Once check_publish_block has confirmed it is time to
+   publish a block, the block header is considered complete, except for the
+   consensus information. The BlockPublisher calls finalize_block with the
+   completed block_header allowing the consensus field to be filled out.
+   Afterwards, the BlockPublisher signs the block and broadcasts it to the
+   network.
+
+This implementation needs to take special care to handle the genesis block
+correctly. During genesis operation, the Consensus.BlockPublisher will be called
+to initialize and finalize a block so that it can be published on the chain
+(see below).
+
+Consensus.BlockVerifier
+-----------------------
+
+The Consenus.BlockVerifier implementation Provides Block verification services
+to the BlockValidator, This gives the consensus algorithm an opportunity to
+check whether the candidate block was published following the consensus rules.
+
+Consensus.ForkResolver
+----------------------
+
+The consensus algorithm is responsible for fork resolution on the system.
+Depending on the consensus algorithm, the determination of the valid block to
+become the chain head will differ. In a Bitcoin Proof of Work consensus, this
+will be the longest chain, whereas PoET uses the measure of aggregate local
+mean (a measure of the total amount of time spent waiting) to determine the
+valid fork. Consensus algorithms with finality, such as PBFT, will only ever
+produce blocks that extend the current head, this function will select the
+block that extends the chain and never the current head.
+
+The ChainController
+===================
+
+The ChainController is responsible for determining which chain the validator is
+currently on and coordinating any change-of-chain activities that need to
+happen. Currently the only listener for chain updates is the BlockPublisher, so
+that it can build a new candidate block on the new chain head.
+
+The ChainController is designed to be able to handle multiple block validation
+activities simultaneously. For instance, if multiple forks form on the network,
+the ChainController can process blocks from all of the competing forks
+simultaneously. This is advantageous as it allows progress to be made even when
+there are several deep forks competing. The current chain can also be advanced
+while a deep fork is being evaluated. This was implemented for cases that could
+happen if a group of validators lost connectivity with the network and later
+rejoined.
+
+Here is the basic flow of the ChainController as a single block is processed. 
+
+.. image:: ../images/journal_chain_controller.*
+   :width: 80%
+   :align: center
+   :alt: Journal Chain Controller Diagram
+
+When a block arrives the ChainController, creates a BlockValidator and
```

Comment:
Comma not needed?
"When a block arrives the **ChainController, creates a** BlockValidator and..."